src/rust/logger/src/backend/file/sender.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use crate::*;
+
+/// A `FileLogSender` implements `Log` and can be registered as the main logger.
+/// At the logging callsite, messages will be filtered and formatted before
+/// being sent over a queue to the corresponding `FileLogReceiver`. To reduce
+/// runtime allocations, a pool of buffers is maintained and recycled for
+/// re-use.
+pub struct FileLogSender {
+    // level filter for determining if a log message should be logged
+    pub(crate) level_filter: LevelFilter,
+    // a queue for submitting log messages to the receiver
+    pub(crate) sender: Queue<Vec<u8>>,
```

Comment:
`sender`->`log_queued`
`buf_pool`->`log_cleared`?

Reply:
`log_tx`, `buf_rx`? would be fine with your suggestions here otherwise

Reply:
I generally want to avoid things like `tx` and `rx` that very much depends on the point of view, unless it's very obvious (e.g. in a server or client). A sender's tx is a receiver's rx, for example. Instead we can maybe focus on the state of these log containers.

But I'm not super opinionated on exactly how to name these, e.g. I'm equally fine with `log_filled` instead of `log_queued` and `log_free` instead of `log_cleared`.